runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -180,35 +180,38 @@ public void setEpoch(long epoch){
      */
     private final CorfuRuntimeParameters parameters;
 
-
-    /**
-     * Flag, if we are connected.
+    /** A {@link CompletableFuture} which is completed when a connection,
+     *  including a successful handshake completes and messages can be sent
+     *  to the remote node.
      */
     @Getter
-    volatile Boolean connected;
+    volatile CompletableFuture<Void> connectionFuture;
```

Comment:
What is the reason for changing this ?

Reply:
Well, first, the connected variable was never properly set. Second, having a future allows a join whenever the status changes.